amp/WEB-INF/src/org/digijava/module/aim/action/CompareActivityVersions.java

```diff
@@ -358,6 +358,8 @@ public ActionForward viewDifferences(ActionMapping mapping, ActionForm form, Htt
             HttpServletResponse response) throws Exception {
 
         CompareActivityVersionsForm vForm = (CompareActivityVersionsForm) form;
+        AmpActivityVersion av = ActivityUtil.loadActivity(vForm.getActivityOneId());
+        vForm.setActivityName(av.getName());
         vForm.setOutputCollectionGrouped(ActivityVersionUtil.compareActivities(vForm.getActivityOneId()));
```

Comment:
the activity is already loaded inside ActivityVersionUtil.compareActivities
you can wrap it inside List<ActivityComparisonResult> and unwrap it in method viewDifferences so you dont have to change the form